stacker/src/shared_arena_hashmap.rs

```diff
@@ -308,6 +312,8 @@ impl SharedArenaHashMap {
         if self.is_saturated() {
             self.resize();
         }
+        // Limit the key size to u16::MAX
+        let key = &key[..std::cmp::min(key.len(), u16::MAX as usize)];
```

Comment:
There is [`impl From<u16> for usize`](https://doc.rust-lang.org/stable/std/primitive.usize.html#impl-From%3Cu16%3E-for-usize), i.e. the cast could be replaced by `.into()`.

Reply:
I wish rust would just coerce types that can be safely cast

here I don't think we gain much for proof reading